### PR TITLE
NLPP house keeping

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Change vrad type back to Real. It is only used as real numbers.
Expand unit tests.

## What type(s) of changes does this code introduce?
- Refactoring (no functional changes, no api changes)
- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

File: src/QMCHamiltonians/NonLocalECPotential.deriv.cpp [hunk 1]
```diff
@@ -85,7 +85,7 @@ NonLocalECPComponent::RealType NonLocalECPComponent::evaluateValueAndDerivatives
     std::fill(dlogpsi_vp.begin(), dlogpsi_vp.end(), 0.0);
     psi.evaluateDerivativesWF(W, optvars, dlogpsi_vp);
     for (int v = 0; v < dlogpsi_vp.size(); ++v)
-      dratio(v, j) = dlogpsi_vp[v];
+      dratio(v, j) = (dlogpsi_vp[v] - dlogpsi[v]);
```
Comment (prckent) [inline src/QMCHamiltonians/NonLocalECPotential.deriv.cpp:88]: Please explain this change

Comment (ye-luo) [reply src/QMCHamiltonians/NonLocalECPotential.deriv.cpp:88]: I will connect https://github.com/QMCPACK/qmcpack/blob/245b753f3096b66724e3cd84770c53f690730c85/src/QMCWaveFunctions/WaveFunctionComponent.h#L580

See the comment line above the function. The subtraction should be inside the function call.
This subtraction is moved from below and then I can make a local branch calling batched and non-batched.

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
Test this please

**Review by prckent — APPROVED** [review]

**Pull request merged**